crates/runtime/src/model/eval/runs.rs

```diff
@@ -0,0 +1,303 @@
+
+pub static EVAL_RUNS_TABLE_TIME_COLUMN: &str = "created_at";
+pub static EVAL_RUNS_TABLE_PRIMARY_KEY: &str = "id";
+pub static EVAL_RUNS_TABLE_SCHEMA: LazyLock<SchemaRef> = LazyLock::new(|| {
+    Arc::new(Schema::new(vec![
+        Field::new(EVAL_RUNS_TABLE_PRIMARY_KEY, DataType::Utf8, false),
+        Field::new(
+            EVAL_RUNS_TABLE_TIME_COLUMN,
+            DataType::Timestamp(TimeUnit::Second, None),
+            false,
+        ),
+        Field::new("dataset", DataType::Utf8, false),
+        Field::new("model", DataType::Utf8, false),
+        Field::new("status", DataType::Utf8, false),
+        Field::new("error_message", DataType::Utf8, true),
+        Field::new("scorers", DataType::new_list(DataType::Utf8, false), false),
+        Field::new(
+            "metrics",
+            DataType::Map(
+                Arc::new(Field::new_struct(
+                    "entries",
+                    vec![
+                        Arc::new(Field::new("keys", DataType::Utf8, false)),
+                        Arc::new(Field::new("values", DataType::Float32, true)),
+                    ],
+                    false,
+                )),
+                false,
+            ),
+            false,
+        ),
+    ]))
+});
+
+/// Add aggregate metrics for an eval run in `spice.evals.runs`.
+/// `metrics` is a map of scorer name to pairs of (metric name, metrics value).
+pub(super) async fn add_metrics_to_eval_run(
+    df: Arc<DataFusion>,
+    id: &EvalRunId,
+    metrics: &HashMap<String, Vec<(String, f32)>>,
+) -> Result<()> {
+    let mut builder = MapBuilder::new(None, StringBuilder::new(), Float32Builder::new());
+    for (scorer, metric_pair) in metrics {
+        for (metric_name, score) in metric_pair {
+            builder
+                .keys()
+                .append_value(format!("{scorer}/{metric_name}"));
+            builder.values().append_value(*score);
+        }
+    }
+
+    builder
+        .append(true)
+        .boxed()
+        .context(FailedToUpdateEvalRunTableSnafu {
+            eval_run_id: id.clone(),
+        })?;
+
+    let mut updates: HashMap<&str, ArrayRef> = HashMap::new();
+    updates.insert("metrics", Arc::new(builder.finish()) as ArrayRef);
+
+    update_eval_run(df, id, updates).await
+}
+
+/// Updates a row in [`EVAL_RUNS_TABLE_REFERENCE`] with the provided status and error message.
+pub async fn update_eval_run_status(
+    df: Arc<DataFusion>,
+    id: &EvalRunId,
+    status: &EvalRunStatus,
+    err_msg: Option<String>,
+) -> Result<()> {
+    let mut updates: HashMap<&str, ArrayRef> = HashMap::new();
+    updates.insert(
+        "status",
+        Arc::new(StringArray::from(vec![status.to_string()])),
+    );
+    if let Some(err) = err_msg {
+        updates.insert("error_message", Arc::new(StringArray::from(vec![err])));
+    };
+    update_eval_run(df, id, updates).await
+}
+
+/// Writes a new row to `spice.evals.runs` table and returns primary key.
+pub async fn start_eval_run(
```

Comment:
```suggestion
pub async fn start_tracing_eval_run(
```

I think this is clearer on what this function actually does